pkg/network/route_cache.go

```diff
@@ -138,6 +140,10 @@ func (c *routeCache) Get(source, dest util.Address, netns uint32) (Route, bool)
 		c.Unlock()
 	}()
 
+	if c.closed {
```

Comment:
Is the issue that we attent to use c.router after c.router has been closed? If so, wouldn't it be cleaner to return `nil, false` whenever c.router.Route() the router is closed?